integration/aggregate_compat_test.go

```diff
@@ -100,7 +103,15 @@ func testAggregateStagesCompat(t *testing.T, testCases map[string]aggregateStage
 					require.NoError(t, targetCursor.All(ctx, &targetRes))
 					require.NoError(t, compatCursor.All(ctx, &compatRes))
 
-					AssertEqualDocumentsSlice(t, compatRes, targetRes)
+					if ignoreOrder {
+						// 1. order of result in compat and target are different because
+						// aggregation can assign BSON array to _id.
```

Comment:
So let's create a separate test if we can't fix array sorting now

Reply:
Sure, let's only test for non array values and fix array sorting in a separate issue!